compiler/noirc_evaluator/src/ssa/ir/map.rs

```diff
@@ -194,6 +194,11 @@ impl<T> DenseMap<T> {
         let ids_iter = (0..self.storage.len() as u32).map(|idx| Id::new(idx));
         ids_iter.zip(self.storage.iter())
     }
+
+    pub(crate) fn rev_iter(&self) -> impl ExactSizeIterator<Item = (Id<T>, &T)> {
+        let ids_iter = (0..self.storage.len() as u32).map(|idx| Id::new(idx));
+        ids_iter.zip(self.storage.iter()).rev()
+    }
```

Comment:
Equivalent to the below no?
```suggestion

    pub(crate) fn rev_iter(&self) -> impl ExactSizeIterator<Item = (Id<T>, &T)> {
        self.iter().rev()
    }
```

Which raises the question of do we need this method?

Reply:
I tried the same, but `DoubleEndedIterator` is not implemented for `ExactSizeIterator` which is needed for `rev`. I will leave a comment

Reply:
Could `self.iter()` and thus DenseMap be changed to return a DoubleEndedIterator instead then?

Reply:
Yeah I could do that as well. I didn't change the type as I wasn't sure if we wanted to keep `ExactSizeIterator`.

Reply:
It doesn't look like it changes much, so I can update the type if we would like

Reply:
I switched to using `DoubleEndedIterator` for code simplicity.